valhalla/baldr/graphreader.h

```diff
@@ -506,6 +524,14 @@ class GraphReader {
     return GetGraphTile(edge->endnode(), tile) ? tile->node(edge->endnode()) : nullptr;
   }
 
+  /**
+   * Method to get the begin node of an edge by using its opposing edges end node
+   * @param edge    the edge whose begin node you want
+   * @param tile    reference to a pointer to a const tile
+   * @return        returns GraphId of begin node of the edge (empty if couldn't find)
+   */
+  GraphId GetBeginNodeId(const DirectedEdge* edge, const GraphTile*& tile);
```

Comment:
maybe put the body of this method in the header as well. as other things start to use it, the importance of the compiler inlining it will increase